### ISSUE TITLE
Passkeys registration fails with "Error! Credential is excluded."

### ISSUE THREAD

**Issue by fgallina** [keepassxreboot/keepassxc-browser#2033]
## Expected Behavior
The passkey is stored sucessfully in KeePassXC

## Current Behavior
The following error message is shown:
![Screenshot from 2023-11-16 11-46-01](https://github.com/keepassxreboot/keepassxc/assets/38927/a4ee9f52-8ac7-4cd0-a78e-00ef3196f0c6)

## Possible Solution
There must be something with the handling of `excludeCredentials` in either the browser plugin or KeePassXC.

## Steps to Reproduce (for bugs)

1. Go to https://passkey.keycloak.ch
2. Click the "Login" link in the top right corner
3. Click "Register" link at the bottom of the sign in form
4. Fill the form, click "Register" 
5. Finally click "Register" in the Security Key Registration form
6. An error message is displayed

## Debug info

Console log with debug mode enabled:
```
[Debug fields.js:167] KeePassXC-Browser - Input fields found: global.js:137:13
Array []
global.js:140:17
[Debug fields.js:167] KeePassXC-Browser - Input fields found: global.js:137:13
Array []
global.js:140:17
[Debug passkeys.js:21] KeePassXC-Browser - publicKey global.js:137:13
Object { attestation: undefined, authenticatorSelection: {…}, challenge: "tjdKrnN8T0O9wcewncTjUg", extensions: undefined, pubKeyCredParams: (1) […], rp: {…}, timeout: undefined, excludeCredentials: [], user: {…} }
global.js:140:17
[Debug keepassxc-browser.js:802] KeePassXC-Browser - Credential is excluded. global.js:137:13
```

KeePassXC - 2.8.0-snapshot (rev 1126055)
KeePassXC-Browser - 1.8.9
Operating system: Linux
Browser: Chrome/Firefox

**Comment by varjolintu** [comment]
Interesting. When trying to login, the site sends only a `challenge` and `rpId` in the public key, but nothing else.

In the register phase there's a parameter `authenticatorSelection: { requireResidentKey: true }`. Sadly, we currently don't support those.

**Comment by varjolintu** [comment]
Let's reopen this. The exceptions or fallbacks needs to be implemented to the extension side.

**Comment by tatroc** [comment]
I am getting this error on Windows 11, keepassxc 2.7.9, Chrome on www.amazon.com. I'm in the account settings page trying to register a passkey. I am unable to do so because of the Error! Credential is excluded.

**Comment by varjolintu** [comment]
@tatroc It means you probably already have a passkey defined for your account and the server puts your user the the excluded credentials list.

**Comment by tatroc** [comment]
@varjolintu Ah thanks, I had one registered under Apple keychain. I deleted the keychain passkey and reregistered it under keypassxc, now it works. Thanks

**Comment by art-shen** [comment]
Have this error with the Arc browser and Google account :'(

**Comment by maurerle** [comment]
I do have this issue with Cisco Duo 2FA Login on Firefox-ESR: duosecurity.com

I do have a working passkey in my keepass from mobile Fennex and KeepassDX though. I wonder if these create issues.

**Comment by varjolintu** [comment]
@maurerle You cannot use the same passkey directly for login? That is why the creation probably fails because you already have a idential passkey.

**Comment by maurerle** [comment]
@varjolintu when creating a passkey I get the error message "Error! Credential is excluded" , when trying to log in though, I get "Error: No logins found".

The Passkey is used as a 2FA after password input and works fine on mobile, but not on Desktop. I suspect the issue to be duo.com - as passkey login works fine with github.com

With debug logs i do get: `challenge: "xxx",enterpriseAttestationPossible: false,extensions: Object { appid: "https://api-12caca12.duosecurity.com" },rpId: "duosecurity.com",timeout: 30000,userVerification: "discouraged",<prototype>: Object { shadowSelector: shadowSelector(), shadowSelectorAll: shadowSelectorAll(), … }`

**Comment by varjolintu** [comment]
@maurerle KeePassXC only checks the exclusion list when registering a new passkey, not when authenticating / logging on.

**Comment by goekce** [comment]
@varjolintu 
I had a similar problem: I wanted to register a new passkey because GitLab showed that none was registered. Then, when I clicked register, I got this error. Without looking at this issue, I would not have found the issue.

I found that I already had an keepassxc entry, even though GitLab showed that I had not registered or set passkeys. There was an interruption during registration, so I ended up with an entry in KeePassXC without completing it.

I would wish that the error message would be more user-friendly, e.g.,

> You are trying to register a new entry, but an entry associated with this website already exists. Maybe a former registration process was interrupted.

If this makes sense, I can happily open a new issue.
